uoapi/timetable/cli.py

```diff
@@ -122,12 +122,25 @@ def cli(args=None):
         ):
             print(json.dumps(out))
 
-def available(retries):
-    tq = qt.TimetableQuery(retries=retries)
-    with tq as gm:
-        out = {"available": list(tq.available.values())}
-    out["messages"] = gm
-    return out
+def available(retries=2):
+    try:
+        tq = qt.TimetableQuery(retries=retries)
+        with tq as gm:
+            out = {"available": [
+                at
+                for at in map(qt.parse_available, tq.available.keys())
+                if at is not None
+            ]}
+        out["messages"] = gm
+        return out
+    except Exception as e:
+        return {
+            "available": [],
+            "messages": [{
+                "type": "error",
+                "message": "Unknown failure: {} = {}".format(type(e), e),
+            }],
+        }
```

Comment:
ideally we could have some more known failures? but that being said if the failures truly are unknown then there's not much to be done

Reply:
So there are plenty of specific error handlers along the way which catch recognized errors.  This is a last-resort catch-all, since the cli isn't really suppose to fail, but rather produce empty output (`"available": []`) with messages saying what happened (`"messages": ...`) (roughly speaking).